src/mudlet.cpp

```diff
@@ -2500,3 +2505,9 @@ void mudlet::requestProfilesToReloadMaps( QList<QString> affectedProfiles )
 {
     emit signal_profileMapReloadRequested( affectedProfiles );
 }
+
+
+// addition to help with dock widget closing.
+void TDockWidget::closeEvent( QCloseEvent * event ) {
+    mudlet::self()->closeWindow( mudlet::self()->getActiveHost(), windowTitle() );
```

Comment:
The active `Host` may not be the same as the one that was originally used to create it. You can test for this (probably) by opening a 2 connections, open a window in one and then close it while the other is selected.

You can store the `Host` in a `QPointer<Host>` in `TDockWidget` and pass that to `closeWindow`.